token/cli/src/main.rs

```diff
@@ -482,6 +483,94 @@ fn command_unwrap(config: &Config, address: Pubkey) -> CommandResult {
     Ok(Some(transaction))
 }
 
+fn command_approve(
+    config: &Config,
+    account: Pubkey,
+    ui_amount: f64,
+    delegate: Pubkey,
+) -> CommandResult {
+    println!(
+        "Approve {} tokens\n  Account: {}\n  Delegate: {}",
+        ui_amount, account, delegate
+    );
+
+    let source_account_balance = config
+        .rpc_client
+        .get_token_account_balance_with_commitment(&account, config.commitment_config)?
+        .value;
+    let source_account = config
+        .rpc_client
+        .get_account_with_commitment(&account, config.commitment_config)?
+        .value
+        .unwrap_or_default();
+    let data = source_account.data.to_vec();
+    let mint_pubkey = Account::unpack_from_slice(&data)?.mint;
+    let amount = spl_token::ui_amount_to_amount(ui_amount, source_account_balance.decimals);
+
+    let mut transaction = Transaction::new_with_payer(
+        &[approve_checked(
+            &spl_token::id(),
+            &account,
+            &mint_pubkey,
+            &delegate,
+            &config.owner.pubkey(),
+            &[],
+            amount,
+            source_account_balance.decimals,
+        )?],
+        Some(&config.fee_payer.pubkey()),
+    );
+
+    let (recent_blockhash, fee_calculator) = config.rpc_client.get_recent_blockhash()?;
+    check_fee_payer_balance(
+        config,
+        fee_calculator.calculate_fee(&transaction.message(), None),
+    )?;
+    let mut signers = vec![config.fee_payer.as_ref(), config.owner.as_ref()];
+    unique_signers!(signers);
+    transaction.sign(&signers, recent_blockhash);
+    Ok(Some(transaction))
+}
+
+fn command_revoke(config: &Config, account: Pubkey) -> CommandResult {
+    let source_account = config
+        .rpc_client
+        .get_account_with_commitment(&account, config.commitment_config)?
+        .value
+        .unwrap_or_default();
+    let data = source_account.data.to_vec();
+    let delegate = Account::unpack_from_slice(&data)?.delegate;
```

Comment:
`get_token_account_with_commitment()` can spare you the data wrangling/unpacking here too